CleverTapSDK/InApps/CTInAppHTMLViewController.m

```diff
@@ -107,12 +109,52 @@ - (void)layoutNotification {
 
 - (void)loadWebView {
     CleverTapLogStaticInternal(@"%@: Loading the web view", [self class]);
+    
+    if (@available(iOS 11.0, *)) {
+        UILayoutGuide *safeArea = self.view.safeAreaLayoutGuide;
+        [NSLayoutConstraint activateConstraints:@[
+            // Use the safe area layout guide to position the view
+            [webView.topAnchor constraintEqualToAnchor: safeArea.topAnchor],
+            [webView.leftAnchor constraintEqualToAnchor: safeArea.leadingAnchor],
+            [webView.rightAnchor constraintEqualToAnchor: safeArea.trailingAnchor],
+            [webView.bottomAnchor constraintEqualToAnchor: safeArea.bottomAnchor]
+        ]];
+    } else {
+        // Fallback on earlier versions
+        [NSLayoutConstraint activateConstraints:@[
+                [webView.topAnchor constraintEqualToAnchor:self.topLayoutGuide.bottomAnchor],
+                [webView.leftAnchor constraintEqualToAnchor:self.view.leadingAnchor],
+                [webView.rightAnchor constraintEqualToAnchor:self.view.trailingAnchor],
+                [webView.bottomAnchor constraintEqualToAnchor:self.bottomLayoutGuide.topAnchor]
+            ]];
+    }
     if (self.notification.url) {
         [webView loadRequest:[NSURLRequest requestWithURL:[NSURL URLWithString:self.notification.url]]];
         webView.navigationDelegate = nil;
     } else{
         [webView loadHTMLString:self.notification.html baseURL:nil];
     }
+    
+    if (self.notification.aspectRatio) {
+        [self updateWebViewForAdvanceInApps];
+    } else {
+        [self updateWebView];
+    }
+}
+
+- (void)updateWebViewForAdvanceInApps {
+    webView.autoresizingMask = UIViewAutoresizingFlexibleBottomMargin | UIViewAutoresizingFlexibleRightMargin | UIViewAutoresizingFlexibleLeftMargin | UIViewAutoresizingFlexibleHeight;
+    
+    if (self.notification.darkenScreen) {
```

Comment:
Can this `backgroundColor` update be moved to `loadWebView`? The code is currently duplicated in `updateWebView` and `updateWebViewForAdvanceInApps`.

Reply:
Done